File: Ozone_Route256_2023/Task_C/solution.py
# (C) Парное программирование (10 баллов), Полное решение: 10 баллов.

def solution():
    devs = int(input())
    skills = list(map(int, input().split()))

    if devs == 2:
        print(1, 2, '\n')
        return

    first = 0
    second = 1

    for _ in range(devs // 2 - 1):
        min_div = abs(skills[first] - skills[second])

        for i in range(first + 2, devs):
            if skills[i] is None:
                continue

            if abs(skills[first] - skills[i]) < min_div:
                min_div = abs(skills[first] - skills[i])
                second = i

            if min_div == 0:
                break
    
        skills[first] = skills[second] = None
        print(first + 1, second + 1)

        first += 1
        while first < devs:
            if not skills[first] is None:
                break
            first += 1
        
        second = first + 1
        while first < devs:
            if not skills[second] is None:
                break
            second += 1

    print(first + 1, second + 1, '\n')


def main():
    for _ in range(int(input())):
        solution()


if __name__ == '__main__':
    main()

File: Ozone_Route256_2023/Task_C/solution.go
// (C) Парное программирование (10 баллов)
// Полное решение: 10 баллов

package main

import "fmt"

func main() {
	var amnt uint8
	fmt.Scanln(&amnt)
	for ; amnt > 0; amnt-- {
		solution()
	}
}

func solution() {
	var devs, first, second, dif, tmp int8
	fmt.Scan(&devs)
	skills := make([]int8, devs)

	for i := 0; i < int(devs); i++ {
		fmt.Scan(&skills[i])
	}

	if devs == 2 {
		fmt.Println(1, 2, "\n")
		return
	}
	second = 1

	for u := devs / 2; u > 1; u-- {
		dif = skills[first] - skills[second]

		if dif < 0 {
			dif = -dif
		}

		for i := first + 2; i < devs; i++ {
			if skills[i] == 0 {
				continue
			}
			tmp = skills[first] - skills[i]
			if tmp < 0 {
				tmp = -tmp
			}
			if tmp < dif {
				dif = tmp
				second = i
			}
			if dif == 0 {
				break
			}
		}
		skills[first] = 0
		skills[second] = 0
		first++
		fmt.Println(first, second+1)

		for ; first < devs; first++ {
			if skills[first] != 0 {
				break
			}
		}
		second = first + 1
		for ; second < devs; second++ {
			if skills[second] != 0 {
				break
			}
		}
	}
	fmt.Println(first+1, second+1, "\n")
}
